MRETestBed/Assets/TestBed Assets/Scripts/Tools/ButtonTool.cs

```diff
@@ -37,16 +37,28 @@ protected override void UpdateTool(InputSource inputSource)
 					var mwUser = buttonBehavior.GetMWUnityUser(inputSource.UserGameObject);
 					if (mwUser != null)
 					{
-						buttonBehavior.Button.StopAction(mwUser);
-						buttonBehavior.Click.StartAction(mwUser);
+						buttonBehavior.Context.EndButton(mwUser, CurrentTargetPoint);
+						buttonBehavior.Context.Click(mwUser, CurrentTargetPoint);
```

Comment:
Have we never checked that the ButtonDown and ButtonUp happened on the same actor before firing click? Seems weird to me.

Reply:
The tools have never been perfect in the test bed.  There is some cleanup that I wanted to do here, but there was the possibility that the MRTK test bed would be the main test bed at some point and all of this code would go away then.

---
In reply to: [434224741](https://github.com/microsoft/mixed-reality-extension-unity/pull/160#discussion_r434224741) [](ancestors = 434224741)